### ISSUE TITLE
REST API issue: product pagination returns products that are ordered internal to the page, but with no overall ordering consistency from page to page

### ISSUE THREAD

**Issue by jakey-wan-kenobi** [woocommerce/woocommerce#16911]
## Prerequisites

- [X ] I have searched for similar issues in both open and closed tickets and cannot find a duplicate
- [X ] The issue still exists against the latest `master` branch of WooCommerce on Github (this is **not** the same version as on WordPress.org!)
- [ X] I have attempted to find the simplest possible steps to reproduce the issue
- [ ] I have included a failing test as a pull request (Optional)

## Steps to reproduce the issue

1. Hit the rest api at /products?per_page=20 (for example, the actual count shouldn't matter)
2. Results returned are order **internally** by menu_order
3. Fetch the remaining pages recursively (for example, like you would do in a scroll-to-load-more-products kind of page)

## Expected/actual behavior

When I follow those steps, I see...
The results of each page are returned (and ordered) correctly, but there's no way to ensure that the first page returned contains all of the first items in the menu_order. 

I was expecting to see...
The first page contains all the products that should come first, and each consecutive page contains products in descending order. This would allow me to only fetch the first 20 products, for example, and only continue to fetch more as the user scrolls, and still have the correct overall product order.

## Isolating the problem

- [ X] This bug happens with only WooCommerce plugin active
- [ ] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/)
- [X ] I can reproduce this bug consistently using the steps above

## WordPress Environment

<details>
```
Copy and paste the system status report from **WooCommerce > System Status** in WordPress admin here.
```
</details>

**Comment by mikejolley** [comment]
`menu_order` is *not* the default for `orderby`. Date is. `menu_order` is not an option.

```
Sort collection by object attribute. Options: date, id, include, title and slug. Default is date.
```

https://woocommerce.github.io/woocommerce-rest-api-docs/#list-all-products

I confirmed this matched up with my results (per_page set to 2 so I could compare more easily), and I confirmed that the store orderby options were not reflected in the API.

Since this is working my side, I can only assume that either an extension or custom code is affecting your API responses, or that the dates are potentially identical in your setup (in which case it may appear random - we've fixed issues like this before in the store itself).

I'll add a fallback to ID for date sorting, but I was not able to replicate your issue.
